tests/bluechi_test/container.py

```diff
@@ -67,8 +67,34 @@ def gather_coverage(self, data_coverage_dir: str) -> None:
         gcno_file_location = "/usr/share/bluechi-coverage/"
         gcda_file_location = "/var/tmp/bluechi-coverage"
 
-        self.get_file(gcda_file_location, data_coverage_dir)
-        self.get_file(gcno_file_location, data_coverage_dir)
+        result, output = self.container.exec_run("rpm -q bluechi-controller")
+        bluechi_version = "bluechi" + str(output.decode()).split("-controller")[1].split("\n")[0]
+        src_file_location = f"/usr/src/debug/{bluechi_version}/src"
+
+        self.container.exec_run(f"cp -r {gcno_file_location}/. {gcda_file_location}")
+
+        result, output = self.container.exec_run(f"find {gcda_file_location} -name '*gcda'")
+        for file in output.split(b"\n"):
+            if b"gcda" in file:
+                file_name_without_hashes = file.split(b"#")[-1].decode()
+                file = gcda_file_location + file.split(b'/var/tmp/bluechi-coverage')[-1].decode()
+                result, output = self.container.exec_run(f"cp {file} {gcda_file_location}/{file_name_without_hashes}")
+                result, output = self.container.exec_run(f"rm {file}")
+
+        self.container.exec_run(f"lcov -c -d {gcda_file_location} --output-file=coverage.info")
+
+        version = bluechi_version.split("bluechi-")[1].split("-0")[0]
+        src = f"/root/rpmbuild/BUILD/bluechi-{version}/src"
+
+        self.container.exec_run(f"mkdir -p {src} ")
+
+        self.container.exec_run(f"cp -r {src_file_location}/. {src}")
+
+        result_file_name = f"res-{self.container.name}"
+
+        self.container.exec_run(f"genhtml coverage.info --output-directory={result_file_name}")
```

Comment:
The above code just generates HTML output for a single test, which is not that useful in a global point of view. Is there a way how to merge those `.info` files and generate global HTML output from all tests?

Reply:
Maybe something like https://stackoverflow.com/a/60636596

Reply:
> The above code just generates HTML output for a single test, which is not that useful in a global point of view.

Wouldn't we want to have both cases? Reports for each test might be useful for error tracking and one combining the results from all as overview.